runtime/compiler/env/j9method.cpp

```diff
@@ -5823,9 +5836,6 @@ TR_J9MethodBase::isUnsafePut(TR::RecognizedMethod rm)
    {
    switch (rm)
       {
-      case TR::sun_misc_Unsafe_compareAndSwapInt_jlObjectJII_Z:
-      case TR::sun_misc_Unsafe_compareAndSwapLong_jlObjectJJJ_Z:
-      case TR::sun_misc_Unsafe_compareAndSwapObject_jlObjectJjlObjectjlObject_Z:
```

Comment:
Was the presence of these three `Unsafe.compareAndSwap` methods in `isUnsafePut` a bug or do the callers of `isUnsafePut` expect this method to return true for those methods?

Reply:
None of the callers of `isUnsafePut` use these three cases. They all have earlier checks for different recognized methods. The three callers are:
`UnsafeFastPath` is trying to turn Unsafe Put into a indirect store node. It doesn't handle CAS operations. An earlier check filters out the CAS recognized methods.
`truncateBooleanForUnsafeGetPut` works with boolean and none of these are booleans. An earlier check limit this to the boolean case which filters out the CAS recognized methods.
`processUnsafeAtomicCall` is only called on the recognized methods: `getAnd[Add|Set][Int|Long]`

I opted to remove the cases since it felt confusing to me. I would expect an "Unsafe Put" to be one of the various Unsafe methods with "Put" in the name that have similar properties. If anything, I would support moving `getAnd[Add|Set][Int|Long]` out of `isUnsafePut` and into its own check since those don't feel like an "Unsafe Put" to me either. But, it does currently work with how the code is written since there are checks that limit a match to the third case.